packages/json/src/lib/JSONProvider.ts

```diff
@@ -88,19 +88,23 @@ export class JSONProvider<StoredValue = unknown> extends JoshProvider<StoredValu
 
   public async [Method.Dec](payload: Payloads.Dec): Promise<Payloads.Dec> {
     const { key, path } = payload;
-    const getPayload = await this[Method.Get]({ method: Method.Get, key, path });
+    const getPayload = await this[Method.Get]({ method: Method.Get, errors: [], key, path });
 
     if (!isPayloadWithData(getPayload)) {
-      return { ...payload, error: this.error({ identifier: CommonIdentifiers.MissingData, method: Method.Dec }, { key, path }) };
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.MissingData, method: Method.Dec }, { key, path }));
+
+      return payload;
     }
 
     const { data } = getPayload;
 
     if (typeof data !== 'number') {
-      return { ...payload, error: this.error({ identifier: CommonIdentifiers.InvalidDataType, method: Method.Dec }, { key, path, type: 'number' }) };
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.InvalidDataType, method: Method.Dec }, { key, path, type: 'number' }));
+
+      return payload;
     }
 
-    await this[Method.Set]({ method: Method.Set, key, path, value: data - 1 });
+    await this[Method.Set]({ method: Method.Set, errors: [], key, path, value: data - 1 });
```

Comment:
What if set returns a payload with errors?

Reply:
I'd say leave it for this PR, but maybe this should be handled later on with every provider.

Reply:
Perhaps open an issue with high severity, cause it's pretty important